ninja-core/src/main/java/ninja/template/directives/TemplateEngineFreemarkerAuthenticityFormDirective.java

```diff
@@ -0,0 +1,42 @@
+package ninja.template.directives;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.util.Map;
+
+import ninja.Context;
+import ninja.utils.NinjaConstant;
+import freemarker.core.Environment;
+import freemarker.template.TemplateDirectiveBody;
+import freemarker.template.TemplateDirectiveModel;
+import freemarker.template.TemplateException;
+import freemarker.template.TemplateModel;
+import freemarker.template.TemplateModelException;
+
+/**
+ * 
+ * @author svenkubiak
+ *
+ */
+@SuppressWarnings("rawtypes")
+public class TemplateEngineFreemarkerAuthenticityFormDirective implements TemplateDirectiveModel {
+    private String authenticityToken;
+
+    public TemplateEngineFreemarkerAuthenticityFormDirective(Context context) {
+        this.authenticityToken = context.getSession().getAuthenticityToken();
```

Comment:
please do this in the execute method. Otherwise you are always adding the __AT to the session - even if it is not needed...